pkg/proxy/proxy.go

```diff
@@ -830,72 +819,54 @@ func (p *Proxy) revertStackUnlocked(revertStack revert.RevertStack) {
 }
 
 // RemoveRedirect removes an existing redirect that has been successfully created earlier.
-func (p *Proxy) RemoveRedirect(id string, wg *completion.WaitGroup) (error, revert.FinalizeFunc, revert.RevertFunc) {
+// Called with 'localEndpoint' passed to 'CreateOrUpdateRedirect' locked for writing!
```

Comment:
We only write so we can clean up some DNS rules; maybe we can get rid of this codepath somehow. I'll think about it. (doesn't need to be part of this PR unless inspiration strikes you)

Reply:
Oh, I think we can move DNS rules caching to its own controller; we should just periodically synchronize them. I'll take this as a separate task.